### DISCUSSION TITLE
Why does the `ThrowsCount` rule ignore `override fun`?

### DISCUSSION THREAD

**Discussion by stellingsimon** [detekt/detekt#6723] [Q&A] [answered]
Thanks for this invaluable piece of software.

[`ThrowsCount`](https://detekt.dev/docs/rules/style/#throwscount) reports this as a violation:

```kotlin
fun f1(x: Int) {
    when (x) {
        1 -> throw IOException()
        2 -> throw IOException()
        3 -> throw IOException()
    }
}
```

but not this:

```kotlin
override fun f3(x: Int) { // does not report overridden function
    when (x) {
        1 -> throw IOException()
        2 -> throw IOException()
        3 -> throw IOException()
    }
}
```

Looking at the [source code of the `ThrowsCount` rule](https://github.com/detekt/detekt/blob/main/detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCount.kt#L56), it seems this behavior is intentional.
At least there's a test case asserting [this exact behavior](https://github.com/detekt/detekt/blob/77b98ea27eba2bb8b8473620e3dfd907dbda5f71/detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCountSpec.kt#L91).
This behavior and the associated test have been there since the rule was added in 2017. 

Still, I wonder: why? A function's body doesn't become more readable just because it's overriding some interface method.
Also, the test's name clearly does not express what it's body implements:

```kotlin
@Test
fun `reports violation by default`() {
    assertThat(subject.lint(code)).hasSize(1)
}
```

So my expectations are met by the test's name, but not the test's body and the current implementation. Which one needs to change? I'm willing to provide a PR once I know what is to be expected here.

**Comment by BraisGabin** [comment]
That's odd. Could you review the git history to see if that was added from the beginning of the rule or if it was added based on an issue request? To me this seems like an issue. I can't imagine any reason to treat the override functions in a different way on this rule.

If there is not issue we could removed. If there is, please link it and we can decide what to do. (probably add a configuration but let's first check it before take any decision)

**Reply by stellingsimon** [reply]
Thank's for the reply.

When `ThrowsCount.kt` was added in 648e5dc0 it already included the 
```
if (!function.hasModifier(KtTokens.OVERRIDE_KEYWORD)) {
```
Its successor commit 259155bbb2e2 then "Added test case for overridden fun in ThrowsCount" with an explicit
```
override fun f3(x: Int) { // does not report overridden function
```
so it looks like an intentional design, but I can't find any hints as to what motivates this.

**Reply by BraisGabin** [reply]
@schalkms do you remember if there was a reason for this? Could we remove it or should we add it behind a configuration?

**Reply by schalkms** [reply]
No, I don’t remember exactly, unfortunately. We had an initiative to exclude overridden functions back then. However, I don’t see a reason here to keep this behavior. Thus, feel free to change this behavior.

**Reply by BraisGabin** [reply, marked as answer]
@stellingsimon then, if you want to go for it just remove the if and fix the tests to reflect this. And thanks for the issue and the research :)

**Reply by stellingsimon** [reply]
Thanks. I reported the issue in #6752 and a proposed fix in #6753.
